src/iuse.cpp

```diff
@@ -2747,160 +2747,542 @@ int iuse::sew(player *p, item *it, bool, point)
-                                 fix->tname().c_str());
-            thread_used = rng(1, 8);
-        } else if (rn <= 8) {
-            p->add_msg_if_player(m_mixed, _("You repair your %s, but waste lots of thread."),
-                                 fix->tname().c_str());
-            if (fix->damage >= 3) {
-                p->consume_items(comps);
+
+            int items_needed = (((mod->volume()) / 3) +1 ); //Add 1 so it isn't zero
+
+            // this will cause issues if/when NPCs start being able to sew.
+            // but, then again, it'll cause issues when they start crafting, too.
+            const inventory &crafting_inv = p->crafting_inventory();
+            bool bFound = false;
+            //go through all discovered repair items and see if we have any of them available
+            for( auto &repair_items_i : repair_items ) {
+                if( crafting_inv.has_amount( repair_items_i, items_needed ) ) {
+                    //we've found enough of a material, use this one
+                    repair_item = repair_items_i;
+                    bFound = true;
+                }
             }
-            fix->damage--;
-            thread_used = rng(1, 8);
-        } else if (rn <= 16) {
-            p->add_msg_if_player(m_good, _("You repair your %s!"), fix->tname().c_str());
-            if (fix->damage >= 3) {
-                p->consume_items(comps);
+            if (!bFound) {
+                for (unsigned int i = 0; i < repair_items.size(); i++) {
+                    p->add_msg_if_player(m_info, _("You don't have enough %s to do that."), plurals[i].c_str());
+                }
+                return 0;
             }
-            fix->damage--;
-        } else {
-            p->add_msg_if_player(m_good, _("You repair your %s completely!"), fix->tname().c_str());
-            if (fix->damage >= 3) {
-                p->consume_items(comps);
+            std::vector<item_comp> comps;
+            comps.push_back(item_comp(repair_item, items_needed));
+
+            p->moves -= 500 * p->fine_detail_vision_mod();
+            p->practice("tailor", 9);
+            int rn = dice(4, 2 + p->skillLevel("tailor"));
+
+            if (p->dex_cur < 8 && one_in(p->dex_cur)) {
+                rn -= rng(2, 6);
+            }
+            if (p->dex_cur >= 16 || (p->dex_cur > 8 && one_in(16 - p->dex_cur))) {
+                rn += rng(2, 6);
+            }
+            if (p->dex_cur > 16) {
+                rn += rng(0, p->dex_cur - 16);
+            }
+
+            if (rn <= 8) {
+                p->add_msg_if_player(m_bad, _("You damage your %s further trying to sew on pockets!"),
+                                     mod->tname().c_str());
+                mod->damage++;
+                if (mod->damage >= 5) {
+                    p->add_msg_if_player(m_bad, _("You destroy it!"));
+                    p->i_rem_keep_contents( pos );
+                }
+            } else
+                if (rn <= 10) {
```

Comment:
You can avoid the additional indentation by using `else if( ... )` like the existing code already does.

Also this contains many repetitions that should be moved into a separate function.